### DISCUSSION TITLE
Device dimensions

### DISCUSSION THREAD

**Discussion by LaurentAjdnik** [pasqal-io/Pulser#160] [Ideas] [answered]
Hi!

`Device.validate_register()` checks a lot of things.

But there is nothing about maximal width/height (and depth in 3D) allowed by a given device for the layout of atoms.

Would it make sense? Or do we consider that they would always "fit" inside the device, whatever the layout is?

**Comment by HGSilveri** [comment, marked as answer]
Hi @LaurentAjdnik ! These restrictions are enforced by the "Maximum distance from origin" constraint of a device, as that's the only relevant limitation. This also means that, in general, you'll want your atom array centered around the origin.

**Reply by LaurentAjdnik** [reply]
Indeed! Damn, I hadn't noticed that `max_radial_distance` constraint... 😵
